#! /bin/sh /usr/share/dpatch/dpatch-run
## 01_perl_example.dpatch by  <octo@collectd.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix example Perl script.

@DPATCH@

diff --git a/src/collectd-perl.pod b/src/collectd-perl.pod
index 6b44722..d5401dd 100644
--- a/src/collectd-perl.pod
+++ b/src/collectd-perl.pod
@@ -656,9 +656,9 @@ A very simple read function might look like:
 
   sub foobar_read
   {
-    my $vl = { plugin => 'foobar' };
+    my $vl = { plugin => 'foobar', type => 'gauge' };
     $vl->{'values'} = [ rand(42) ];
-    plugin_dispatch_values ('gauge', $vl);
+    plugin_dispatch_values ($vl);
     return 1;
   }
 
